apps/meteor/app/cloud/server/functions/getCheckoutUrl.ts

```diff
@@ -0,0 +1,39 @@
+import { serverFetch as fetch } from '@rocket.chat/server-fetch';
+
+import { SystemLogger } from '../../../../server/lib/logger/system';
+import { settings } from '../../../settings/server';
+
+type CheckoutBodyParams = {
+	okCallback: string;
+	cancelCallback: string;
+	meta?: Record<string, string>;
+};
+
+export const getCheckoutUrl = async (token: string, body: CheckoutBodyParams) => {
+	try {
+		const billingUrl = settings.get<string>('Cloud_Billing_Url');
+
+		const response = await fetch(`${billingUrl}/api/v2/checkout`, {
+			method: 'POST',
+			headers: {
+				Authorization: `Bearer ${token}`,
+			},
+			body,
+		});
+
+		if (!response.url) {
+			const { i18n } = await response.json();
+			throw new Error(i18n);
+		}
```

Comment:
not sure this check is doing what you expect.. may you please explain what exactly is going on here?